File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.py
def solution(begin, end):
    answer = []
    for i in range(begin, end+1):
        temp = []
        if i==1:
            answer.append(0)
            continue
        for j in range(2,int(i**(1/2))+1):
            if i%j==0 :
                if i//j>10**7:
                    temp.append(j)
                    continue
                else:
                    answer.append(i//j)
                    break
        else:
            if len(temp) > 0:
                answer.append(temp[len(temp)-1])
            else:
                answer.append(1)
    return answer

File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.js
function solution(begin, end) {
    const answer = [];
    
    const isPrime = (num) => {
        if (num <= 1) return false;
        if (num === 2) return true;
        const sqrt = Math.sqrt(num);
        for (let i = 2; i <= sqrt; i++) {
            if (num % i === 0) return false;
        }
        return true;
    };
    
    const get = (num) => {
        if (num === 1) return 0
        if (isPrime(num)) return 1
        
        const temp = [];
        
        for (let i = 2; i <= Math.sqrt(num); i++) {
            if (num % i === 0) {  // num이 i로 나누어 떨어진다면 (즉, i가 약수라면)
                if (num / i <= 10000000) {
                    return num / i
                }
                else temp.push(i)
            }
        }
        
        return temp.at(-1);
    }

    for (let i = begin; i <= end; i++) {
        const num = get(i);
        answer.push(num);
    }

    return answer;
}